metadata.pl

```diff
@@ -0,0 +1,273 @@
+   hpc.job.padcswan.jobenv hpc.job.prep15.jobenv hpc.job.prep15.subshellpids
+   hpc.job.prep15.nodelist hpc.job.padcswan.jobenv hpc.job.padcswan.subshellpids 
+   hpc.job.padcswan.nodelist post.opendap.files post.opendap.lsu_tds.linkablehosts
+   post.opendap.lsu_tds.copyablehosts
+);
+# the following properties from run.properties are array valued and designated
+# in run.properties as comma separated and should be stored as an array in json
+my @comma_properties = qw( notification.email.activate_list
+   notification.email.new_advisory_list notification.email.post_init_list
+   notification.email.job_failed_list notification.hpc.email.notifyuser
+   notification.opendap.email.opendapnotify 
+);
+# the following properties from run.properties are stored as individual
+# top level properties but should be considered a sublist of related 
+# metadata with a list of properties associated with them in turn
+# (to start off with, " File Name" and " Format")
+# adcirc.file.output.list
+my @outputfile_properties = ( 
+    "Water Surface Elevation Stations", "Water Surface Elevation",
+    "Water Current Velocity", "Barometric Pressure Stations",
+    "Wind Velocity Stations", "Barometric Pressure", "Wind Velocity",
+    "Maximum Water Surface Elevation", "Maximum Current Speed",
+    "Maximum Wind Speed", "Minimum Barometric Pressure",
+    "Maximum Wave Radiation Stress", "Mean Wave Direction",
+    "Maximum Mean Wave Direction", "Significant Wave Height",
+    "Maximum Significant Wave Height", "Mean Wave Period",
+    "Maximum Mean Wave Period", "Peak Wave Period",
+    "Maximum Peak Wave Period", "Initially Dry", "Inundation Time",
+    "Maximum Inundation Depth", "Ever Dried", "End Rising Inundation",
+    "Wind Velocity 10m Stations", "Wind Velocity 10m",
+    "Maximum Wind Speed 10m",
+);
+
+our $this = "metadata.pl"; 
+#
+GetOptions(
+           "metadatafile=s" => \$metadatafile,    
+           "keys=s" => \$keys,
+           "mapsacalar=s" => \$mapscalar,
+           "jsonify" => \$jsonify           
+          );
+# open metadata file
+if ($metadatafile eq "null") { 
+   &stderrMessage("ERROR","Did not provide the name of the metadata file.");
+   exit 1;
+}    
+# remove a trailing slash (if any, just in case)
+if ( substr($metadatafile,-1,1) eq "/" ) {
+    chop($metadatafile);
+}
+# determine the file type by grabbing all characters from last dot to end of line
+my ($type) = $metadatafile =~ /(\.[^.]+)$/;
+# determine path to the metadata file (if any was provided)
+my $dirpath = "";
+$last_slash = rindex($metadatafile,"/");
+if ( $last_slash != -1 ) {
+    $dirpath = substr($metadatafile,0,$last_slash);
+}
+#
+#  J S O N
+if ( $type eq ".json" ) {
+    unless (open(F,"<$metadatafile")) {
+        &stderrMessage("ERROR","Failed to open '$metadatafile': $!.");
+        die;
+    }
+    # slurp the file contents into a scalar variable
+    $file_content = do { local $/; <F> };
+    close(F);        
```

Comment:
The slurp is actually pretty close to a good idiom I just recently learned. Update to use `$F`, and use `scalar` to ensure proper variable context:

```
$file_content = do { local $/; scalar <$F> };
close $F;
```